test/plugin/base_test.rb

```diff
@@ -1434,9 +1433,41 @@ def test_k8s_monitored_resource_fallback
           local_resource_id: RANDOM_LOCAL_RESOURCE_ID),
         # When 'kube-env' is present, "compute.googleapis.com/resource_name" is
         # not added.
-        expected_params: COMPUTE_PARAMS
-      },
-      # Specific cases for k8s_node.
+        expected_params: COMPUTE_PARAMS_WITH_CONTAINER_TAG
+      }
+    ].each do |test_params|
+      new_stub_context do
```

Comment:
Do we want to create a helper method for this section to avoid duplicated code?

Reply:
+1 on the helper method.

Reply:
A new helper method is a bit overkill and will decrease readability. I refactored the existing helper methods a bit. Hope that helps. PTAL.

Reply:
This helps a bit, but Bryan and I were both talking about the code that follows stub setup, which is identical modulo the tag passed into `create_driver`...

Reply:
Good point.